File: packages/lib/neft-screen-brightness/native/ios/ScreenBrightnessExtension.swift
import UIKit

fileprivate let app = App.getApp()
fileprivate let binding = NativeBinding("ScreenBrightness")

fileprivate class BrightnessHandler {
    var timer: Timer?
    let screen = UIScreen.main

    init() {
        NotificationCenter.default.addObserver(self, selector: #selector(handleBrightnessDidChange(notfication:)), name: UIScreen.brightnessDidChangeNotification, object: nil)
    }

    @objc func handleBrightnessDidChange(notfication: NSNotification) {
        pushBrightness()
    }

    func pushBrightness() {
        binding.pushEvent("brightness", args: [screen.brightness])
    }

    func setBrightness(_ val: CGFloat) {
        screen.brightness = val
    }
}

extension Extension.ScreenBrightness {
    fileprivate static let handler = BrightnessHandler()

    static func register() {
        binding
            .onCall("getBrightness") { (args: [Any?]) in
                handler.pushBrightness()
            }
            .onCall("setBrightness") { (args: [Any?]) in
                handler.setBrightness((args[0] as! Number).float())
            }
            .finalize()
    }
}

File: packages/lib/neft-screen-brightness/native/android/ScreenBrightnessExtension.kt
package io.neft.extensions.screenbrightness_extension

import io.neft.App
import io.neft.client.NativeBinding

object ScreenBrightnessExtension {
    private val APP = App.getInstance()
    private val BINDING = NativeBinding("ScreenBrightness")

    @JvmStatic
    fun register() {
        BINDING
                .onCall("getBrightness") {
                    pushBrightness()
                }
                .onCall("setBrightness") {
                    value ->
                    setBrightness((value[0] as Number).toFloat())
                    pushBrightness()
                }
    }

    private fun pushBrightness() {
        BINDING.pushEvent("brightness", getBrightness())
    }

    fun getBrightness() : Float {
        return APP.activity.window.attributes.screenBrightness
    }

    fun setBrightness(value: Float) {
        val window = APP.activity.window
        val attributes = window.attributes
        attributes.screenBrightness = value
        window.attributes = attributes
    }
}
